app/src/main/java/com/example/androidapp/DatabaseController.java

```diff
@@ -167,6 +173,75 @@ public static void updateUser(Context context, User user) {
 
     }
 
+
+    public static void insertStep(String s, String day, String hour, Context context) {
+        ContentValues values = new ContentValues();
+        values.put("timestamp", s);
+        values.put("day", day);
+        values.put("hour", hour);
+        DatabaseController databaseHelper = new DatabaseController(context);
+        SQLiteDatabase database = databaseHelper.getReadableDatabase();
+        database.insert("Step",null,values);
+    }
+
+    public static void loadSteps(Context context){
+        List<String> dates = new LinkedList<String>();
+        DatabaseController databaseHelper = new DatabaseController(context);
+        SQLiteDatabase database = databaseHelper.getReadableDatabase();
+
+        Cursor cursor = database.query("Step", new String [] {"timestamp"}, null, null, "timestamp",
+                null, null );
+
+        cursor.moveToFirst();
+        for (int index=0; index < cursor.getCount(); index++){
+            dates.add(cursor.getString(0));
+            cursor.moveToNext();
+        }
+        database.close();
+
+    }
+
+
+    public static Integer loadSingleStep(Context context, String date){
+        List<String> steps = new LinkedList<String>();
+        DatabaseController databaseHelper = new DatabaseController(context);
+        SQLiteDatabase database = databaseHelper.getReadableDatabase();
+
+        Cursor cursor = database.query("Step", null, "day = ?", new String [] {date}, null,
+                null, null );
+
+        cursor.moveToFirst();
+        for (int index=0; index < cursor.getCount(); index++){
+            steps.add(cursor.getString(0));
+            cursor.moveToNext();
+        }
+        database.close();
+        Integer numSteps = steps.size();
+        return numSteps;
+    }
+    public static Map<String, Integer> loadStepsByDay(Context context){
```

Comment:
What does exactly do this method? Does it return the steps of a given day grouped by hour or does it perform the same thing of loadSingleSteps() but in a different way?

Reply:
(However comment all the methods shortly describing what they do)